src/hotspot/share/c1/c1_GraphBuilder.cpp

```diff
@@ -1890,6 +1893,15 @@ void GraphBuilder::invoke(Bytecodes::Code code) {
     default:
       break;
     }
+
+    // Additional receiver subtype checks for interface calls via invokespecial or invokeinterface.
+    if (receiver_constraint != nullptr) {
+      int index = state()->stack_size() - (target->arg_size_no_receiver() + 1);
+      Value receiver = state()->stack_at(index);
+      CheckCast* c = new CheckCast(receiver_constraint, receiver, copy_state_before());
+      c->set_incompatible_class_change_check();
```

Comment:
set_incompatible_class_change_check() seems OK for invokeinterface, but for invokespecial, I believe the interpreter throws IllegalAccessError.

Reply:
Thanks to point it out.  

I take a look at invokespecial and invokeinterface part of JVM spec. There are multiple cases. I think I better use uncommon trap and let interpreter to cover them all. It's not a hot path anyway. 
 
By reverting this code, `java/lang/invoke/SpecialInterfaceCall.java` is fixed.